configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java

```diff
@@ -275,13 +280,25 @@ public int deleteExpiredRemoteSessions(Clock clock, Duration expiryTime) {
             if (session.getStatus() == Session.Status.ACTIVATE) continue;
             if (sessionHasExpired(session.getCreateTime(), expiryTime, clock)) {
                 log.log(Level.FINE, "Remote session " + sessionId + " for " + tenantName + " has expired, deleting it");
-                session.delete();
+                deleteSession(session);
                 deleted++;
             }
         }
         return deleted;
     }
 
+
+    public void deactivate(RemoteSession remoteSession) {
+        remoteSessionCache.addSession(remoteSession.deactivate());
```

Comment:
`addSession` should perhaps also be named `putSession`?

Reply:
Agreed, will fix